src/foam/dao/index/TreeIndex.java

```diff
@@ -136,6 +136,8 @@ public SelectPlan planSelect(Object state, Sink sink, long skip, long limit, Com
     Object[] statePredicate = simplifyPredicate(state, predicate);
     state = statePredicate[0];
     predicate = (Predicate) statePredicate[1];
+    if ( predicate == null && sink instanceof GroupBy )
```

Comment:
Also check that it's the correct property in the GroupBy. Also, simplify/eliminate the comparator before creating the GroupByPlan.